source/exe/posix/platform_checks.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <fcntl.h>
+#include <sys/resource.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "envoy/filesystem/filesystem.h"
+
+#include "common/common/assert.h"
+
+namespace Envoy {
+
+static void check_sysctl_long(std::string path, long expected, Filesystem::Instance& file_system) {
```

Comment:
Pass `path` by `absl::string_view`. If there was a good reason to pass by `std::string` then you'd use a const ref.

Pass 'expected' as an int64_t. Use absl::SimpleAtoI<int64_t> for the conversion.